packages/atclient/src/atkeys.c

```diff
@@ -681,48 +764,81 @@ exit: { return ret; }
 int atclient_atkeys_populate_from_path(atclient_atkeys *atkeys, const char *path) {
   int ret = 1;
 
-  atclient_atkeysfile atkeysfile;
-  atclient_atkeysfile_init(&atkeysfile);
+  atclient_atkeysfile atkeys_file;
+  atclient_atkeysfile_init(&atkeys_file);
+
+  if ((ret = atclient_atkeysfile_from_path(&atkeys_file, path)) != 0) {
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
+                 "atclient_atkeysfile_from_path: %d | failed to read file at path: %s\n", ret, path);
+    goto exit;
+  }
+
+  if ((ret = atclient_atkeys_populate_from_atkeys_file(atkeys, &atkeys_file)) != 0) {
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
+                 "atclient_atkeys_populate_from_atkeys_file: %d | failed to decrypt & populate struct \n", ret);
+    goto exit;
+  }
+
+  goto exit;
+exit: {
+  atclient_atkeysfile_free(&atkeys_file);
+  return ret;
+}
+}
+
+int atclient_atkeys_populate_from_string(atclient_atkeys *atkeys, const char *file_string) {
```

Comment:
new populate variant, similar to populate from path, but is canonical to the atclient_atkeys_file_from_string variant for atkeys_file